src/compas_eve/mqtt/mqtt_paho.py

```diff
@@ -87,8 +87,11 @@ def subscribe(self, topic, callback):
         subscribe_id = "{}:{}".format(event_key, id(callback))
 
         def _local_callback(msg):
-            msg = topic._message_from_json(msg.payload.decode())
-            callback(msg)
+            decoded_message = self.codec.decode(msg.payload)
+            callback(decoded_message)
+
+            # msg = topic._message_from_json(msg.payload.decode())
+            # callback(msg)
```

Comment:
Handling decoding needs to follow the same approach as the one commented on the encoding, because there's this custom type parsing supported in there, so, we need to change `Topic._message_from_json` to this:
```python
    def _message_from_data(self, data):
        """Converts a decoded data object back into a message instance."""
        return self.message_type.parse(data)
```

and this needs to change a bit as well:
```suggestion
            decoded_data = self.codec.decode(msg.payload.decode())
            decoded_message = topic._message_from_data(decoded_data)
            callback(decoded_message)
```

Reply:
and the `mqtt_cli.py` would change as well to this sequence of calls